---
title: "Analysis of data from "
author: "Veronika Niederlova"
date: "2022-10-16"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(include = TRUE, warning = FALSE, message = FALSE, error = TRUE, cache = TRUE)

library("Matrix")
library(Seurat)
library(DT)
library(dplyr)
library(here)
library(ggplot2)
library(SingleCellExperiment)
library(scater)
library(mclust)
library(kableExtra)
library(cowplot)
library(tidyverse)
library(reshape)
library(ProjecTILs)
library(STACAS)
library(TILPRED)
library(AUCell)
library(reshape2)
library(grid)
library(gridExtra)
library(SingleR)

library(scales)
library(ggthemes)
library(RColorBrewer)
display.brewer.all(n = NULL, type = "all", select = NULL,
                   colorblindFriendly = FALSE)

convertHumanGeneList <- function(x){

require("biomaRt")
human = useMart("ensembl", dataset = "hsapiens_gene_ensembl")
mouse = useMart("ensembl", dataset = "mmusculus_gene_ensembl")

genesV2 = getLDS(attributes = c("hgnc_symbol"), filters = "hgnc_symbol", values = x , mart = human, attributesL = c("mgi_symbol"), martL = mouse, uniqueRows=T)

humanx <- unique(genesV2[, 2])

# Print the first 6 genes found to the screen
print(head(humanx))
return(humanx)
}

 reduced_annot  <- function(seurat_object, tier_name, n_cells_to_other){
    md  <- seurat_object@meta.data
    tier_name_red  <- paste0(tier_name,"_red")
    cell_types_keep  <- (md  %>% group_by_at(tier_name)  %>% tally()  %>% arrange(desc(n)) %>% dplyr::filter(n > n_cells_to_other))[[tier_name]]
    md[[tier_name_red]]  <- if_else(md[[tier_name]] %in% cell_types_keep,as.character(md[[tier_name]]),"Other")
    seurat_object  <- AddMetaData(seurat_object, metadata = md[[tier_name_red]], tier_name_red)
    return(seurat_object)
}

```



Read data from well 3 - Spleen

```{r}
seu <- readRDS("~/PRIMUS/data/48_lab/Project scRNAseq/DATA_scRNAseq/Experiments/Exp11/04_Raw_data/Rds/exp11W3_init.rds")

seu[["percent.mt"]] <- PercentageFeatureSet(seu, pattern = "^mt-")
seu[["percent.rt"]] <- PercentageFeatureSet(seu, pattern = "^Rp[ls]")

# Remove TRAV, TRBV genes: blood young B6
rownames(seu)
seu_annots <- seu@meta.data
seu_counts <- seu[!(grepl('Trav', rownames(seu)) |
          grepl('Trbv', rownames(seu)) |
          grepl('mt-', rownames(seu)) |
            grepl('Rpl', rownames(seu)) |
          grepl('Rps', rownames(seu)) |
          grepl('MGP-', rownames(seu)) 
            ),]

seu <- CreateSeuratObject(counts = seu_counts[["RNA"]]@counts, project = "seu", min.cells = 3, min.features = 200, meta.data = seu_annots)
 seu <- NormalizeData(object = seu)
  seu <- ScaleData(seu, verbose = FALSE)
  seu <- FindVariableFeatures(seu, selection.method = "vst", nfeatures = 1000, 
                                       verbose = FALSE)
  seu <- RunPCA(seu, npcs = 20, verbose = FALSE)
  seu <- RunUMAP(seu, reduction = "pca", dims = 1:20)
  seu <- FindNeighbors(seu, dims = 1:20)
  seu <- FindClusters(seu, resolution = 0.9)

 counts <- GetAssayData(seu[["RNA"]])

seu_singler <- CreateSinglerObject(counts=counts,
  project.name= "E07_Tregs", # choose
  min.genes = 200, # ignore cells with fewer than 200 transcripts
  technology = "10x", # choose
  species = "Mouse",
  citation = "Our 10x data exp 01-05", # choose
  normalize.gene.length = FALSE,        # needed for full-length platforms (e.g. smartseq)
  variable.genes = "de",  # see vignette
  fine.tune = FALSE, # TRUE would take very long
  reduce.file.size = TRUE, # leave out less-often used fields 
  do.signatures = FALSE,
  do.main.types = TRUE,
  numCores = 16)


seu <- AddMetaData(seu,
seu_singler$singler[[1]]$SingleR.single$labels,
                                 col.name = "Immgen_annot_single")
seu <- AddMetaData(seu,
seu_singler$singler[[1]]$SingleR.single.main$labels,
                                 col.name = "Immgen_annot_single_main")
seu <- AddMetaData(seu,
seu_singler$singler[[2]]$SingleR.single.main$labels,
                                 col.name = "MouseRNAseq_single_main")
seu <- AddMetaData(seu,
seu_singler$singler[[2]]$SingleR.single$labels,
                                 col.name = "MouseRNAseq_single")

DimPlot(seu)
#ggsave(filename = paste0("funda_quick/Init_processing/_Dimplot.png"), width = 20, height = 13,units = "cm")

cowplot::plot_grid(DimPlot(seu, group.by = "MouseRNAseq_single"),
                   DimPlot(seu, group.by = "MouseRNAseq_single_main"),
    DimPlot(seu, group.by = "Immgen_annot_single_main"),
    DimPlot(seu, group.by = "Immgen_annot_single"), ncol = 2)

#ggsave(filename = paste0("funda_quick/Init_processing/_Annotations.png"), width = 30, height = 20,units = "cm")


FeaturePlot(seu, features = c("Mki67","Ccl5","Ccr7","Malat1"))
#ggsave(filename = paste0("funda_quick/Init_processing/_Features.png"), width = 30, height = 20,units = "cm")

VlnPlot(seu, features = c("nCount_RNA","nFeature_RNA","percent.mt","percent.rt"))
#ggsave(filename = paste0("funda_quick/Init_processing/_QC_metrics.png"), width = 30, height = 20,units = "cm")

DimPlot(seu, group.by = "Immgen_annot_single_main", label = T)

seu_spleen <- seu
```


### Add hashtags

```{r}
seu_spleen$hashtags %>% table

md_spleen <- seu_spleen@meta.data 
md_spleen <- md_spleen %>% mutate(MouseID = str_replace_all(string = hashtags, pattern = "_Hashtag", replacement = ""))
md_spleen <- md_spleen %>% mutate(Diet = if_else(MouseID %in% c("Mouse_1", "Mouse_2", "Mouse_3", "Mouse_4", "Mouse_5"), "GFD", "Standard"))

seu_spleen@meta.data <- md_spleen

DimPlot(seu_spleen, group.by = "Diet")
```

## Filter high quality & lymphoid populations

```{r}
VlnPlot(seu_spleen, features = c("nCount_RNA","nFeature_RNA","percent.mt","percent.rt"))
seu_spleen_filt <- subset(seu_spleen, nFeature_RNA > 750 & percent.mt < 7.5 & Immgen_annot_single_main %in% c("T cells", "NK cells", "NKT", "Tgd", "ILC"))


seu_spleen
seu_spleen_filt

seu_spleen_filt <- NormalizeData(object = seu_spleen_filt)
  seu_spleen_filt <- ScaleData(seu_spleen_filt, verbose = FALSE)
  seu_spleen_filt <- FindVariableFeatures(seu_spleen_filt, selection.method = "vst", nfeatures = 500, 
                                       verbose = FALSE)
  seu_spleen_filt <- RunPCA(seu_spleen_filt, npcs = 12, verbose = FALSE)
  seu_spleen_filt <- RunUMAP(seu_spleen_filt, reduction = "pca", dims = 1:12)
  seu_spleen_filt <- FindNeighbors(seu_spleen_filt, dims = 1:12)
  seu_spleen_filt <- FindClusters(seu_spleen_filt, resolution = 0.7)

  DimPlot(seu_spleen_filt, group.by = "Immgen_annot_single_main", label = T, repel = T) 
  DimPlot(seu_spleen_filt, group.by = "MouseRNAseq_single_main", label = T, repel = T) 
  
  DimPlot(seu_spleen_filt, group.by = "Immgen_annot_single", label = T, repel = T) 
  DimPlot(seu_spleen_filt, label = T, repel = T) 
  
  
  FeaturePlot(seu_spleen_filt, features = c("Cd3d", "Trdc", "Trgc1", "Trac", "Zbtb16b", "Cd160", "Klrb1", "Klrk1",
                                            "Cd4", "Cd8a", "Cd8b1", "Ccl5", "Ccr7", "Foxp3"))
  
  FeaturePlot(seu_spleen_filt, features = c("Il2ra", "Foxp3", "Gata3", "Rorc", "Ifng", "Mki67", 
                                            "Tbx21", "Eomes", "Egr1", "Tnfrsf9", "Izumo1r", "Ctla4"))
  FeaturePlot(seu_spleen_filt, features = c("Klrg1", "Cx3cr1", "Myc", "Myb", "Pdcd1", "Havcr2", 
                                            "Tim3", "Lag3", "Sca1", "Kit", "Il7r", "Ikzf2"))
  FeaturePlot(seu_spleen_filt, features = c("Cxcr5", "Bcl6", "Cd69", "Cd27", "Cd28", "B3gat",
                                            "Isg15", "Itga4", "Il2rb", "Pdrm1"))
  
  
  DimPlot(seu_spleen_filt, group.by = "Diet", label = T, repel = T) 
  DimPlot(seu_spleen_filt)
```


```{r}

```


Read data from well 4 - pLN

```{r}
seu <- readRDS("~/PRIMUS/data/48_lab/Project scRNAseq/DATA_scRNAseq/Experiments/Exp11/04_Raw_data/Rds/exp11W4_init.rds")

seu[["percent.mt"]] <- PercentageFeatureSet(seu, pattern = "^mt-")
seu[["percent.rt"]] <- PercentageFeatureSet(seu, pattern = "^Rp[ls]")

# Remove TRAV, TRBV genes: blood young B6
rownames(seu)
seu_annots <- seu@meta.data
seu_counts <- seu[!(grepl('Trav', rownames(seu)) |
          grepl('Trbv', rownames(seu)) |
          grepl('mt-', rownames(seu)) |
            grepl('Rpl', rownames(seu)) |
          grepl('Rps', rownames(seu)) |
          grepl('MGP-', rownames(seu)) 
            ),]

seu <- CreateSeuratObject(counts = seu_counts[["RNA"]]@counts, project = "seu", min.cells = 3, min.features = 200, meta.data = seu_annots)
 seu <- NormalizeData(object = seu)
  seu <- ScaleData(seu, verbose = FALSE)
  seu <- FindVariableFeatures(seu, selection.method = "vst", nfeatures = 1000, 
                                       verbose = FALSE)
  seu <- RunPCA(seu, npcs = 20, verbose = FALSE)
  seu <- RunUMAP(seu, reduction = "pca", dims = 1:20)
  seu <- FindNeighbors(seu, dims = 1:20)
  seu <- FindClusters(seu, resolution = 0.9)

 counts <- GetAssayData(seu[["RNA"]])

seu_singler <- CreateSinglerObject(counts=counts,
  project.name= "E07_Tregs", # choose
  min.genes = 200, # ignore cells with fewer than 200 transcripts
  technology = "10x", # choose
  species = "Mouse",
  citation = "Our 10x data exp 01-05", # choose
  normalize.gene.length = FALSE,        # needed for full-length platforms (e.g. smartseq)
  variable.genes = "de",  # see vignette
  fine.tune = FALSE, # TRUE would take very long
  reduce.file.size = TRUE, # leave out less-often used fields 
  do.signatures = FALSE,
  do.main.types = TRUE,
  numCores = 16)


seu <- AddMetaData(seu,
seu_singler$singler[[1]]$SingleR.single$labels,
                                 col.name = "Immgen_annot_single")
seu <- AddMetaData(seu,
seu_singler$singler[[1]]$SingleR.single.main$labels,
                                 col.name = "Immgen_annot_single_main")
seu <- AddMetaData(seu,
seu_singler$singler[[2]]$SingleR.single.main$labels,
                                 col.name = "MouseRNAseq_single_main")
seu <- AddMetaData(seu,
seu_singler$singler[[2]]$SingleR.single$labels,
                                 col.name = "MouseRNAseq_single")

DimPlot(seu)
#ggsave(filename = paste0("funda_quick/Init_processing/_Dimplot.png"), width = 20, height = 13,units = "cm")

#ggsave(filename = paste0("funda_quick/Init_processing/_Annotations.png"), width = 30, height = 20,units = "cm")


FeaturePlot(seu, features = c("Mki67","Ccl5","Ccr7","Malat1"))
#ggsave(filename = paste0("funda_quick/Init_processing/_Features.png"), width = 30, height = 20,units = "cm")

VlnPlot(seu, features = c("nCount_RNA","nFeature_RNA","percent.mt","percent.rt"))
#ggsave(filename = paste0("funda_quick/Init_processing/_QC_metrics.png"), width = 30, height = 20,units = "cm")

DimPlot(seu, group.by = "Immgen_annot_single_main", label = T)
FeaturePlot(seu, features = c("Cd8a","Cd4"))
FeaturePlot(seu, features = c("Sell","Foxp3"))
FeaturePlot(seu, features = c("Egr1","Tnfrsf9"))

seu_pLN <- seu
```


## Add mouse information

```{r}
md_pLN <- seu_pLN@meta.data 
md_pLN <- md_pLN %>% mutate(MouseID = str_replace_all(string = hashtags, pattern = "_Hashtag", replacement = ""))
md_pLN <- md_pLN %>% mutate(Diet = if_else(MouseID %in% c("Mouse_1", "Mouse_2", "Mouse_3", "Mouse_4", "Mouse_5"), "GFD", "Standard"))

seu_pLN@meta.data <- md_pLN

DimPlot(seu_pLN, group.by = "MouseID")



VlnPlot(seu_pLN, features = c("nCount_RNA","nFeature_RNA","percent.mt","percent.rt"))

DimPlot(seu_pLN, group.by = "Immgen_annot_single_main", label = T, repel = T) 
DimPlot(seu_pLN, group.by = "MouseRNAseq_single_main", label = T, repel = T) 
  
  
seu_pLN_filt <- subset(seu_pLN, nFeature_RNA > 750 & percent.mt < 7.5 & Immgen_annot_single_main %in% c("T cells", "NK cells", "NKT", "Tgd", "ILC"))


seu_pLN
seu_pLN_filt

seu_pLN_filt <- NormalizeData(object = seu_pLN_filt)
  seu_pLN_filt <- ScaleData(seu_pLN_filt, verbose = FALSE)
  seu_pLN_filt <- FindVariableFeatures(seu_pLN_filt, selection.method = "vst", nfeatures = 500, 
                                       verbose = FALSE)
  seu_pLN_filt <- RunPCA(seu_pLN_filt, npcs = 12, verbose = FALSE)
  seu_pLN_filt <- RunUMAP(seu_pLN_filt, reduction = "pca", dims = 1:12)
  seu_pLN_filt <- FindNeighbors(seu_pLN_filt, dims = 1:12)
  seu_pLN_filt <- FindClusters(seu_pLN_filt, resolution = 0.7)

  DimPlot(seu_pLN_filt, group.by = "Immgen_annot_single_main", label = T, repel = T) 
  DimPlot(seu_pLN_filt, group.by = "MouseRNAseq_single_main", label = T, repel = T) 
  
  DimPlot(seu_pLN_filt, group.by = "Immgen_annot_single", label = T, repel = T) 
  DimPlot(seu_pLN_filt, label = T, repel = T) 
  
  
  FeaturePlot(seu_pLN_filt, features = c("Cd3d", "Trdc", "Trgc1", "Trac", "Zbtb16b", "Cd160", "Klrb1", "Klrk1",
                                            "Cd4", "Cd8a", "Cd8b1", "Ccl5", "Ccr7", "Foxp3"))
  
  FeaturePlot(seu_pLN_filt, features = c("Il2ra", "Foxp3", "Gata3", "Rorc", "Ifng", "Mki67", 
                                            "Tbx21", "Eomes", "Egr1", "Tnfrsf9", "Izumo1r", "Ctla4"))
  FeaturePlot(seu_pLN_filt, features = c("Klrg1", "Cx3cr1", "Myc", "Myb", "Pdcd1", "Havcr2", 
                                            "Tim3", "Lag3", "Sca1", "Kit", "Il7r", "Ikzf2"))
  FeaturePlot(seu_pLN_filt, features = c("Cxcr5", "Bcl6", "Cd69", "Cd27", "Cd28", "B3gat",
                                            "Isg15", "Itga4", "Il2rb", "Pdrm1"))
  
  
  DimPlot(seu_pLN_filt, group.by = "Diet", label = T, repel = T) 
  DimPlot(seu_pLN_filt)
```


## Merged

```{r}

seu_spleen_filt$Tissue <- "Spleen"
seu_pLN_filt$Tissue <- "pLN"

 funda_merge <- merge(seu_spleen_filt, seu_pLN_filt)
 funda_merge <- NormalizeData(object = funda_merge)
  funda_merge <- ScaleData(funda_merge, verbose = FALSE)
  funda_merge <- FindVariableFeatures(funda_merge, selection.method = "vst", nfeatures = 1000, 
                                       verbose = FALSE)
  funda_merge <- RunPCA(funda_merge, npcs = 10, verbose = FALSE)
  funda_merge <- RunUMAP(funda_merge, reduction = "pca", dims = 1:10)
  funda_merge <- FindNeighbors(funda_merge, dims = 1:10)
  funda_merge <- FindClusters(funda_merge, resolution = 1.1)
  DimPlot(funda_merge, label = T)
  

  DimPlot(funda_merge, group.by = "Tissue") 
  DimPlot(funda_merge, group.by = "Diet") 
  
  
  FeaturePlot(funda_merge, features = c("Cd3d", "Trdc", "Trgc1", "Trac", "Cd160", "Klrb1", "Klrk1",
                                            "Cd4", "Cd8a", "Cd8b1", "Ccl5", "Ccr7", "Foxp3"))
  
  FeaturePlot(funda_merge, features = c("Il2ra", "Foxp3", "Gata3", "Rorc", "Ifng", "Mki67", 
                                            "Tbx21", "Eomes", "Egr1", "Tnfrsf9", "Izumo1r", "Ctla4"))
  
  FeaturePlot(funda_merge, features = c("Klrg1", "Cx3cr1", "Myc", "Myb", "Pdcd1", "Havcr2", 
                                            "Tim3", "Lag3", "Sca1", "Kit", "Il7r", "Ikzf2", "Zbtb16", "Rora"))
  
  FeaturePlot(funda_merge, features = c("Cxcr5", "Bcl6", "Cd69", "Cd27", "Cd28", "B3gat",
                                            "Isg15", "Itga4", "Il2rb", "Pdrm1"))
  
  

```


```{r}
DimPlot(funda_merge, group.by = "Immgen_annot_single_main", label = T, repel = T) 
  DimPlot(funda_merge, group.by = "MouseRNAseq_single_main", label = T, repel = T) 
  
  DimPlot(funda_merge, group.by = "Immgen_annot_single", label = T, repel = T) 
  DimPlot(funda_merge, label = T, repel = T) 
  
  
  reduced_annot  <- function(seurat_object, tier_name, n_cells_to_other){
    md  <- seurat_object@meta.data
    tier_name_red  <- paste0(tier_name,"_red")
    cell_types_keep  <- (md  %>% group_by_at(tier_name)  %>% tally()  %>% arrange(desc(n)) %>% dplyr::filter(n > n_cells_to_other))[[tier_name]]
    md[[tier_name_red]]  <- if_else(md[[tier_name]] %in% cell_types_keep,as.character(md[[tier_name]]),"Other")
    seurat_object  <- AddMetaData(seurat_object, metadata = md[[tier_name_red]], tier_name_red)
    return(seurat_object)
}

  
  funda_merge <- reduced_annot(funda_merge, "Immgen_annot_single", n_cells_to_other = 30)
funda_merge$Immgen_annot_single_red %>% table
DimPlot(funda_merge, group.by = "Immgen_annot_single_red", label = T, repel = T) 
 
```


## Annotations

```{r}
md <- funda_merge@meta.data

md <- md %>% mutate(annotations_fine = recode(seurat_clusters, 
                                                     "0" = "00 CD4 Naive 1",
                                                     "1" = "01 CD8 Naive 1",
                                                     "2" = "02 CD4 Naive 2",
                                                     "3" = "03 CD4 Naive 3",
                                                     "4" = "04 CD8 Naive 2",
                                                     "5" = "05 CD4 Naive 4",
                                                     "6" = "06 CD4 Naive 5",
                                                     "7" = "07 CD4 Treg 1",
                                                     "8" = "08 CD4 Treg 2",
                                                     "9" = "09 Tgd and NKT ",
                                                     "10" = "10 NK cells",
                                                     "11" = "11 CD4 Tfh",
                                                     "12" = "12 CD4 Egr Myb",
                                                     "13" = "13 CD8 ISAGhi",
                                                     "14" = "14 CD4 Early act. 41BB CD69",
                                                     "15" = "15 CD8 AIMT",
                                                     "16" = "16 Proliferating",
                                                     "17" = "17 ILC",
                                                     "18" = "18 CD8 Egr"))

funda_merge@meta.data <- md
DimPlot(funda_merge, group.by = "annotations_fine")
```

```{r}
saveRDS(seu_pLN_filt, "data/seu_pLN_filt_nod_mapping.rds")
saveRDS(seu_spleen_filt, "data/seu_spleen_filt_nod_mapping.rds")
saveRDS(funda_merge, "data/seu_merged_nod_mapping.rds")
```

```{r}
funda_merge <- readRDS("data/seu_merged_nod_mapping.rds")
seu_pLN_filt <- readRDS("data/seu_pLN_filt_nod_mapping.rds")
seu_spleen_filt <- readRDS("data/seu_spleen_filt_nod_mapping.rds")

DimPlot(funda_merge, label = T)
 DimPlot(funda_merge, group.by = "Immgen_annot_single_main", label = T, repel = T) 
 DimPlot(funda_merge, group.by = "annotations_fine", label = T, repel = T) 
 
```

```{r}
funda_merge <- BuildClusterTree(
  funda_merge,
  dims = 1:10,
  reorder = FALSE,
  reorder.numeric = FALSE
)

tree <- funda_merge@tools$BuildClusterTree
tree$tip.label <- levels(funda_merge$annotations_fine)

tree$tip.label

p <- ggtree::ggtree(tree, aes(x, y)) +
  scale_y_reverse() +
  ggtree::geom_tree() +
  ggtree::theme_tree() +
  ggtree::geom_tiplab(offset = 1) +
  ggtree::geom_tippoint(color = (scales::hue_pal() (19)), shape = 16, size = 5) +
  coord_cartesian(clip = 'off') +
  theme(plot.margin = unit(c(0,2.5,0,0), 'cm'))

p
#ggsave('plots/cluster_tree.png', p, height = 4, width = 6)
```

## Cluster markers

```{r}
mrk <- FindAllMarkers(funda_merge, logfc.threshold = log(1.5), min.diff.pct = 0.05)

mrk
```



## Cluster abundance boxplot

```{r}

seurat_meta_data <- funda_merge@meta.data

# Create grouped dataframe, calculate the frequencies of clusters
df4 <- seurat_meta_data %>% 
  mutate(MouseID_tissue = paste(MouseID, Tissue)) %>% 
  group_by(MouseID_tissue, annotations_fine) %>% 
  summarise(n = n()) %>% 
  unique() %>% 
  mutate(freq = n / sum(n)) 

# Control - all sums should be one
df4 %>% group_by(MouseID_tissue) %>% summarise(sum = sum(freq))

# As we've lost non-grouping variables, let's join them back
md_to_join <- seurat_meta_data %>% 
  dplyr::select(MouseID, Diet, Tissue) %>% 
  unique() %>% 
  mutate(MouseID_tissue = paste(MouseID, Tissue))
df4  <- left_join(df4, md_to_join)

df4
# The final plot

df4 %>% 
  mutate(Tissue_Diet = paste(Tissue, Diet)) %>% 
  ggplot(aes(x = Tissue_Diet, y = freq)) + # you can change the x to whatever variable you're interested in
  geom_boxplot(outlier.shape = NA) + 
  geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + 
  geom_jitter(position=position_jitter(0.2), size = 2, aes(color = Diet)) + # in aes, you can also use shape or fill (for the shapes that allow it)
  facet_wrap(~annotations_fine, scales = "free") +
  ylab("Frequency") +
  xlab("Condition") +
  theme_classic() +
  theme(strip.background = element_blank(), panel.grid = element_blank()) + 
  theme(axis.text.x = element_text(angle = 90))

ggsave("fig/clusters_boxplots.png", width = 25, height = 25, units = "cm")


df4 %>% 
  mutate(Tissue_Diet = paste(Tissue, Diet)) %>% 
  ggplot(aes(x = Tissue_Diet, y = freq)) + # you can change the x to whatever variable you're interested in
  geom_boxplot(outlier.shape = NA) + 
  geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + 
  geom_jitter(position=position_jitter(0.2), size = 2, aes(color = Diet)) + # in aes, you can also use shape or fill (for the shapes that allow it)
  facet_wrap(~annotations_fine, scales = "free") +
  ylab("Frequency") +
  xlab("Condition") +
  theme_classic() +
  theme(strip.background = element_blank(), panel.grid = element_blank()) + 
  theme(axis.text.x = element_text(angle = 90))


for(i in levels(factor(df4$annotations_fine))) {
    df_filt  <- df4 %>% filter(annotations_fine == i)
    print(paste(i, kruskal.test(as.numeric(df_filt$freq) ~ df_filt$Diet)$p.value))}
```

## DE genes

CD4 naive standard vs gluten free
```{r}

```

CD8 naive standard vs gluten free

```{r}

```

## Testing FACS results

Animals on gluten free diet have:
* higher CD44 on B cells, CD4 T cells and CD8 T cells
* higher CD62L on AE CD8 T cells
* lower CX3CR1 on AE CD8 T cells
* lower CD49d on AE CD8 T cells
* higher CD69 on AE CD8 T cells (although the expression is nearly zero, so careful here)
* lower percentage of CD44+ CX3CR1+ CD8 T cells out of AE CD8 T cells
* lower percentage of CX3CR1+ CD8 T cells out of AE CD8 T cells
* lower percentage of KLRK1+ CD8 T cells out of AE CD8 T cells
* higher percentage of AIMT cells
* lower percentage of CD137+ cells out of AIMT CD8+ cells (although the percentage in general is very low so be careful with conclusions)
* lower percentage ox CD44+ CX3CR1+ and CD44+ KLRK1+ T cells out of all CD8 T cells (in pLN)
* higher CD44 on Naive CD8 T cells
* higher frequency of central memory CD8 T cells (CD62L+ CD44+)
* higher frequency of CD4+ TCRgd T cells
* lower frequency of CX3CR1+ CD44+ TCRgd T cells in iLN
* lower frequency of CX3CR1+ and KLRK1 on TCRgd T cells in iLN
* lower frequency of CD62L- CD44- TCRgd T cells

```{r}
gene_list <- stringr::str_to_sentence(c("KLRK1", "KLRC1", "KLRC2", "KLRC3", "KLRD1", "NCR2", "NCR1"))

seu_merged <- AddModuleScore(
  object = funda_merge,
  features = list(c(gene_list)),
  search = T,
  assay = "RNA",
  name = 'KEGG_NK_cytotoxicity')


FeaturePlot(seu_merged, features = "KEGG_NK_cytotoxicity1") 
VlnPlot(seu_merged, features = "KEGG_NK_cytotoxicity1", pt.size = 0, group.by = "MouseID")

ggtheme <- function() {
  theme(
    axis.text = element_text(size = 20),
    axis.title = element_text(size = 20),
    text = element_text(size = 20, colour = "black"),
    legend.text = element_text(size = 20),
    legend.key.size =  unit(10, units = "points")
    
  )
}

ggplot(data.frame(Score = seu_merged$KEGG_NK_cytotoxicity1,
                  MouseID = seu_merged$MouseID), 
       aes(x = reorder(MouseID, Score, FUN = median), y = Score)) +
  geom_violin() + 
  stat_summary(fun = "median",
               geom = "crossbar", 
               width = 0.5,
               colour = "red") + 
  ggtitle("NK cytotoxicity signature") + 
  ggtheme() + 
  theme(panel.background = element_blank(), 
        axis.text.x = element_text(angle = 45, hjust = 1))

```
# Unconventional populations

```{r}
DimPlot(funda_merge, label = T)

unconv_prolif <- subset(funda_merge, seurat_clusters %in% c(7,9,10,16,17))

 unconv_prolif <- NormalizeData(object = unconv_prolif)
  unconv_prolif <- ScaleData(unconv_prolif, verbose = FALSE)
  unconv_prolif <- FindVariableFeatures(unconv_prolif, selection.method = "vst", nfeatures = 1000, 
                                       verbose = FALSE)
  unconv_prolif <- RunPCA(unconv_prolif, npcs = 10, verbose = FALSE)
  unconv_prolif <- RunUMAP(unconv_prolif, reduction = "pca", dims = 1:10)
  unconv_prolif <- FindNeighbors(unconv_prolif, dims = 1:10)
  unconv_prolif <- FindClusters(unconv_prolif, resolution = 1.1)
  DimPlot(unconv_prolif, label = T)
  

  DimPlot(unconv_prolif, group.by = "Tissue") 
  DimPlot(unconv_prolif, group.by = "Diet") 
  


DimPlot(unconv_prolif, group.by = "MouseID") + ylim(c(-2,3)) + xlim(c(5,8))
DimPlot(unconv_prolif, group.by = "MouseID") + ylim(c(-2,3)) + xlim(c(5,8))


FeaturePlot(unconv_prolif, features = c("Mki67", "Trgc1", "Kit", "Il23r"))

FeaturePlot(unconv_prolif, features = c("Cd4", "Cd8a", "Mki67", "Rorc"))
FeaturePlot(unconv_prolif, features = c("Cd3d", "Il1r1", "Cd160", "Ncr1"))
FeaturePlot(unconv_prolif, features = c("Foxp3", "Il2ra", "Trdc", "Trgv2"))
FeaturePlot(unconv_prolif, features = c("Gata3", "Tbx21", "Ifng", "Bcl6"))
FeaturePlot(unconv_prolif, features = c("Ikzf2", "Klrk1", "Klrd1", "Il7r"))

```

```{r}


  
  unconv_prolif <- reduced_annot(unconv_prolif, "Immgen_annot_single", n_cells_to_other = 20)
unconv_prolif$Immgen_annot_single_red %>% table
DimPlot(unconv_prolif, group.by = "Immgen_annot_single_red", label = T, repel = T) 
 
ggsave("fig/unconventional_dimplot.png", width = 35, height = 25, units = "cm")
```

Boxplots

```{r}
seurat_meta_data <- unconv_prolif@meta.data

# Create grouped dataframe, calculate the frequencies of clusters
df4 <- seurat_meta_data %>% 
  mutate(MouseID_tissue = paste(MouseID, Tissue)) %>% 
  group_by(MouseID_tissue, seurat_clusters) %>% 
  summarise(n = n()) %>% 
  unique() %>% 
  mutate(freq = n / sum(n)) 

# Control - all sums should be one
df4 %>% group_by(MouseID_tissue) %>% summarise(sum = sum(freq))

# As we've lost non-grouping variables, let's join them back
md_to_join <- seurat_meta_data %>% 
  dplyr::select(MouseID, Diet, Tissue) %>% 
  unique() %>% 
  mutate(MouseID_tissue = paste(MouseID, Tissue))
df4  <- left_join(df4, md_to_join)

df4
# The final plot

df4 %>% 
  mutate(Tissue_Diet = paste(Tissue, Diet)) %>% 
  ggplot(aes(x = Tissue_Diet, y = freq)) + # you can change the x to whatever variable you're interested in
  geom_boxplot(outlier.shape = NA) + 
  geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + 
  geom_jitter(position=position_jitter(0.2), size = 2, aes(color = Diet)) + # in aes, you can also use shape or fill (for the shapes that allow it)
  facet_wrap(~seurat_clusters, scales = "free") +
  ylab("Frequency") +
  xlab("Condition") +
  theme_classic() +
  theme(strip.background = element_blank(), panel.grid = element_blank()) + 
  theme(axis.text.x = element_text(angle = 90))

ggsave("fig/clusters_boxplots_unconventional.png", width = 25, height = 25, units = "cm")
```


# Clonal expansion

## Clone abundance
```{r}
metadata_6 <- unconv_prolif@meta.data
rownames(metadata_6) <- colnames(unconv_prolif)

metadata_6 <- metadata_6 %>% mutate(
  clone_nt = paste(cdr3_B_nt,cdr3_A1_nt,cdr3_A2_nt),
  clone_aa = paste("CDR3b",cdr3_B,"CDR3a",cdr3_A1)
)

clone_table <- metadata_6 %>%
  dplyr::group_by(factor(clone_aa)) %>%
          dplyr::summarize(n = n(), sum = sum()) %>%
    arrange(desc(n))

clone_table
```


```{r}
metadata_6$test <- 0
metadata_6 <- metadata_6 %>% group_by(test, clone_aa) %>% mutate(clone_abundance = as.numeric(n()))


metadata_6 <- as.data.frame(metadata_6 %>% mutate(clone_abundance = as.numeric(ifelse(clone_abundance==1069,NA_integer_,clone_abundance))) %>%
  mutate(log_clone_abundance = log(clone_abundance, base = 2)))
rownames(metadata_6) <- colnames(funda_merge)

funda_merge@meta.data <- metadata_6
rownames(funda_merge@meta.data) <- colnames(funda_merge)

funda_merge <- AddMetaData(funda_merge, as.numeric(metadata_6$clone_abundance), "clone_abundance")
rownames(funda_merge@meta.data) <- colnames(funda_merge)

FeaturePlot(funda_merge, reduction = "umap", features = "log_clone_abundance")

FeaturePlot(subset(funda_merge, clone_abundance >=1 & clone_abundance < 1069), reduction = "umap", features = "log_clone_abundance", cols = c("lightblue","firebrick")) + theme_classic() + theme(plot.title = element_text(hjust = 0.5)) + ggtitle("Log2 clone abundance")
#ggsave("fig/Log2CloneAbundance.png", width = 13, height = 8, units = "cm", dpi = 120)
#ggsave("fig/Log2CloneAbundance.svg", width = 13, height = 8, units = "cm", dpi = 120)

DimPlot(funda_merge, label = T)

# Clone abundance bar graph
metadata_6 <- metadata_6 %>%
  mutate( clone_abundance_group = case_when(clone_abundance==5 ~ "5",
                                            clone_abundance==4 ~ "4",
                                            clone_abundance==3 ~ "3",
                                            clone_abundance==2 ~ "2",
                                            clone_abundance==1 ~ "1",
                                            TRUE ~ "1"
                                            ))

metadata_6 %>% 
   ggplot(aes(x = seurat_clusters)) + geom_bar(aes(fill = factor(clone_abundance_group)), position = "fill") + coord_flip() +
  scale_fill_brewer(palette = "Blues") + xlab("Frequency")+ylab("") + theme_classic() + theme(plot.title = element_text(hjust = 0.5), legend.position="right", panel.border = element_blank(), legend.title = element_blank()) + ggtitle("Clone abundance in clusters")


ggsave("fig/Log2CloneAbundance_bar.png", width = 18, height = 10, units = "cm", dpi = 120)
ggsave("fig/Log2CloneAbundance_bar.svg", width = 18, height = 10, units = "cm", dpi = 120)

```

# Clone abundance = boxplot 
```{r}


df4 <-metadata_6 %>% filter(Tissue == "spleen") %>% mutate(animal_age = paste(Sample_ID, age), clon_expanded = if_else(clone_abundance_group == 1,"Unique clone","2+ clones")) %>% group_by(animal_age, clon_expanded) %>% 
  summarise(n = n()) %>%
  mutate(freq = n / sum(n)) 

df4 %>% group_by(animal_age) %>% summarise(sum = sum(freq))

    df4 %>%
   mutate(Age = case_when(grepl("Young",animal_age) ~ "Young",
                             grepl("Old",animal_age) ~ "Old"
                             )) %>% 
     ggplot(aes(x = factor(Age, levels = c("Young","Old")), y = freq*100)) + 
 geom_boxplot(outlier.shape = NA) +
 geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + geom_jitter(position=position_jitter(0.2), size = 2, aes(color = Age)) + scale_color_manual(values= c("firebrick","lightskyblue3")) + theme_minimal() + facet_wrap(~clon_expanded, scales = "free", ncol = 2) 
  ggsave("results/BoxPlot_clon_expanded.png", width = 20, height = 8, units = "cm", dpi = 550)

  
```

# CD8 and CD4 separately

1. Remove unconventional T cells

```{r}
DimPlot(funda_merge, label = T)
convT <- subset(funda_merge, Immgen_annot_single_main == "T cells" & seurat_clusters %in% c(0:8,11:18))

 convT <- NormalizeData(object = convT)
  convT <- ScaleData(convT, verbose = FALSE)
  convT <- FindVariableFeatures(convT, selection.method = "vst", nfeatures = 1000, 
                                       verbose = FALSE)
  convT <- RunPCA(convT, npcs = 10, verbose = FALSE)
  convT <- RunUMAP(convT, reduction = "pca", dims = 1:10)
  convT <- FindNeighbors(convT, dims = 1:10)
  convT <- FindClusters(convT, resolution = 1.1)
  DimPlot(convT, label = T)
  

  DimPlot(convT, group.by = "Tissue", label = T, repel = T) 
  DimPlot(convT, group.by = "Diet", label = T, repel = T) 
  
  
  FeaturePlot(convT, features = c("Cd3d", "Trdc", "Trgc1", "Trac", "Cd160", "Klrb1", "Klrk1",
                                            "Cd4", "Cd8a", "Cd8b1", "Ccl5", "Ccr7", "Foxp3"))
 
  which(rownames(convT) == "Cd4")
  cd4_expression <- convT@assays$RNA@counts[4934,] > 0
  cd8a_expression <- convT@assays$RNA@counts[4658,] > 0
  cd8b_expression <- convT@assays$RNA@counts[4654,] > 0
  
  md <- convT@meta.data
  md$cd4_expression <- cd4_expression
  md$cd8a_expression <- cd8a_expression
  md$cd8b_expression <- cd8b_expression

  md <- md %>% mutate(cd4_or_cd8 = if_else(cd4_expression == F & cd8a_expression == T & cd8b_expression == T, 
                                           "CD8", 
                                           if_else(cd4_expression == T & cd8a_expression == F & cd8b_expression == F, "CD4","Unclear")))  

convT@meta.data <- md

DimPlot(convT, group.by = "cd4_or_cd8", label = T)


DimPlot(convT, group.by = "cd4_or_cd8", split.by = "seurat_clusters", ncol = 4)
  
 md <- md %>% mutate(cd4_or_cd8_2 = if_else(cd4_expression == F & cd8a_expression == T & cd8b_expression == T, 
                                           "CD8", 
                                           if_else(cd4_expression == T & cd8a_expression == F & cd8b_expression == F, "CD4",if_else(seurat_clusters %in% c(0,1,3,4,6,7,8,9,12,13,14,15), "CD4", "CD8"))))  

convT@meta.data <- md

DimPlot(convT, group.by = "cd4_or_cd8_2", label = T)

```
## CD8

```{r}
cd8 <- subset(convT, cd4_or_cd8_2 == "CD8")

DimPlot(cd8)

 cd8 <- NormalizeData(object = cd8)
  cd8 <- ScaleData(cd8, verbose = FALSE)
  cd8 <- FindVariableFeatures(cd8, selection.method = "vst", nfeatures = 1000, 
                                       verbose = FALSE)
  cd8 <- RunPCA(cd8, npcs = 10, verbose = FALSE)
  cd8 <- RunUMAP(cd8, reduction = "pca", dims = 1:10)
  cd8 <- FindNeighbors(cd8, dims = 1:10)
  cd8 <- FindClusters(cd8, resolution = 0.3)
  DimPlot(cd8, label = T)
  DimPlot(cd8, label = T, group.by = "Diet")
  DimPlot(cd8, label = T, group.by = "Tissue")
  
```

### CD8 DE spleen

### Seurat

```{r}
cd8_spl <- subset(cd8, Tissue == "Spleen")
Idents(cd8_spl) <- cd8_spl$Diet
de_cd8_spl <- FindAllMarkers(cd8_spl, only.pos = T)
de_cd8_spl

```

### DeSeq

```{r}
sum_exp_cd8_spl <- AggregateExpression(cd8_spl, group.by = "MouseID", slot = "counts")

md <- data.frame(Diet = c("GF", "GF", "GF", "GF", "GF", "STD", "STD", "STD", "STD"))

# Create a DESeq dataset
library(DESeq2)
dds_cd8_spl <- DESeqDataSetFromMatrix(countData = sum_exp_cd8_spl$RNA,
                              colData = md,
                              design= ~ Diet) 

rownames(dds_cd8_spl) <- rownames(sum_exp_cd8_spl$RNA)

# Run the DESeq2 algorithm, which will give us the differentially expressed genes
dds_cd8_spl <- DESeq(dds_cd8_spl)

plotCounts(dds_cd8_spl, gene="Cxcl10", intgroup="Diet")
plotCounts(dds_cd8_spl, gene="Gzmk", intgroup="Diet")

## PCA
pca_cd8_spl = vst(dds_cd8_spl, blind = FALSE)
pca_cd8_spl_plot <- plotPCA(pca_cd8_spl, intgroup = c("Diet"))
pca_cd8_spl_plot

## Results data frame
res <- results(dds_cd8_spl)
results <- as.data.frame(dplyr::mutate(as.data.frame(res), sig=ifelse(res$padj<0.05, "FDR<0.05", "Not Sig")), row.names=rownames(res))

results %>% arrange(desc(log2FoldChange))
results %>% arrange(log2FoldChange)

```


### CD8 DE pLN

### Seurat

```{r}
cd8_pLN <- subset(cd8, Tissue == "pLN")
Idents(cd8_pLN) <- cd8_pLN$Diet
de_cd8_pLN <- FindAllMarkers(cd8_pLN, only.pos = T)
de_cd8_pLN

```

### DeSeq

```{r}
sum_exp_cd8_pLN <- AggregateExpression(cd8_pLN, group.by = "MouseID", slot = "counts")

md <- data.frame(Diet = c("GF", "GF", "GF", "GF", "GF", "STD", "STD", "STD", "STD"))

# Create a DESeq dataset
library(DESeq2)
dds_cd8_pLN <- DESeqDataSetFromMatrix(countData = sum_exp_cd8_pLN$RNA,
                              colData = md,
                              design= ~ Diet) 

rownames(dds_cd8_pLN) <- rownames(sum_exp_cd8_pLN$RNA)

# Run the DESeq2 algorithm, which will give us the differentially expressed genes
dds_cd8_pLN <- DESeq(dds_cd8_pLN)

plotCounts(dds_cd8_pLN, gene="Cx3cr1", intgroup="Diet")
plotCounts(dds_cd8_pLN, gene="Gzmk", intgroup="Diet")
plotCounts(dds_cd8_pLN, gene="Cd44", intgroup="Diet")
plotCounts(dds_cd8_pLN, gene="Klrk1", intgroup="Diet")

## PCA
pca_cd8_pLN = vst(dds_cd8_pLN, blind = FALSE)
pca_cd8_pLN_plot <- plotPCA(pca_cd8_pLN, intgroup = c("Diet"))
pca_cd8_pLN_plot

## Results data frame
res <- results(dds_cd8_pLN)
results <- as.data.frame(dplyr::mutate(as.data.frame(res), sig=ifelse(res$padj<0.05, "FDR<0.05", "Not Sig")), row.names=rownames(res))

results %>% arrange(desc(log2FoldChange))
results %>% arrange(log2FoldChange)

```


## CD4

```{r}
cd4 <- subset(convT, cd4_or_cd8_2 == "CD4")

DimPlot(cd4)

 cd4 <- NormalizeData(object = cd4)
  cd4 <- ScaleData(cd4, verbose = FALSE)
  cd4 <- FindVariableFeatures(cd4, selection.method = "vst", nfeatures = 1000, 
                                       verbose = FALSE)
  cd4 <- RunPCA(cd4, npcs = 10, verbose = FALSE)
  cd4 <- RunUMAP(cd4, reduction = "pca", dims = 1:10)
  cd4 <- FindNeighbors(cd4, dims = 1:10)
  cd4 <- FindClusters(cd4, resolution = 0.5)
  DimPlot(cd4, label = T)
  DimPlot(cd4, label = T, group.by = "Diet")
  DimPlot(cd4, label = T, group.by = "Tissue")
  
```

### CD4 DE spleen

### Seurat

```{r}
cd4_spl <- subset(cd4, Tissue == "Spleen")
Idents(cd4_spl) <- cd4_spl$Diet
de_cd4_spl <- FindAllMarkers(cd4_spl, only.pos = T)
de_cd4_spl

```

### DeSeq

```{r}
sum_exp_cd4_spl <- AggregateExpression(cd4_spl, group.by = "MouseID", slot = "counts")

md <- data.frame(Diet = c("GF", "GF", "GF", "GF", "GF", "STD", "STD", "STD", "STD"))

# Create a DESeq dataset
library(DESeq2)
dds_cd4_spl <- DESeqDataSetFromMatrix(countData = sum_exp_cd4_spl$RNA,
                              colData = md,
                              design= ~ Diet) 

rownames(dds_cd4_spl) <- rownames(sum_exp_cd4_spl$RNA)

# Run the DESeq2 algorithm, which will give us the differentially expressed genes
dds_cd4_spl <- DESeq(dds_cd4_spl)

plotCounts(dds_cd4_spl, gene="Cxcl10", intgroup="Diet")
plotCounts(dds_cd4_spl, gene="Gzmk", intgroup="Diet")

## PCA
pca_cd4_spl = vst(dds_cd4_spl, blind = FALSE)
pca_cd4_spl_plot <- plotPCA(pca_cd4_spl, intgroup = c("Diet"))
pca_cd4_spl_plot

## Results data frame
res <- results(dds_cd4_spl)
results <- as.data.frame(dplyr::mutate(as.data.frame(res), sig=ifelse(res$padj<0.05, "FDR<0.05", "Not Sig")), row.names=rownames(res))

results %>% arrange(desc(log2FoldChange))
results %>% arrange(log2FoldChange)

```


### CD4 DE pLN

### Seurat

```{r}
cd4_pLN <- subset(cd4, Tissue == "pLN")
Idents(cd4_pLN) <- cd4_pLN$Diet
de_cd4_pLN <- FindAllMarkers(cd4_pLN, only.pos = T)
de_cd4_pLN

```

### DeSeq

```{r}
sum_exp_cd4_pLN <- AggregateExpression(cd4_pLN, group.by = "MouseID", slot = "counts")

md <- data.frame(Diet = c("GF", "GF", "GF", "GF", "GF", "STD", "STD", "STD", "STD"))

# Create a DESeq dataset
library(DESeq2)
dds_cd4_pLN <- DESeqDataSetFromMatrix(countData = sum_exp_cd4_pLN$RNA,
                              colData = md,
                              design= ~ Diet) 

rownames(dds_cd4_pLN) <- rownames(sum_exp_cd4_pLN$RNA)

# Run the DESeq2 algorithm, which will give us the differentially expressed genes
dds_cd4_pLN <- DESeq(dds_cd4_pLN)

plotCounts(dds_cd4_pLN, gene="Cxcl10", intgroup="Diet")
plotCounts(dds_cd4_pLN, gene="Ifng", intgroup="Diet")

## PCA
pca_cd4_pLN = vst(dds_cd4_pLN, blind = FALSE)
pca_cd4_pLN_plot <- plotPCA(pca_cd4_pLN, intgroup = c("Diet"))
pca_cd4_pLN_plot

## Results data frame
res <- results(dds_cd4_pLN)
results <- as.data.frame(dplyr::mutate(as.data.frame(res), sig=ifelse(res$padj<0.05, "FDR<0.05", "Not Sig")), row.names=rownames(res))

results %>% arrange(desc(log2FoldChange))
results %>% arrange(log2FoldChange)

```